src/diff/children.js

```diff
@@ -69,38 +69,25 @@ export function diffChildren(
 		// or we are rendering a component (e.g. setState) copy the oldVNodes so it can have
 		// it's own DOM & etc. pointers
 		else if (typeof childVNode == 'string' || typeof childVNode == 'number') {
-			childVNode = newParentVNode._children[i] = createVNode(
-				null,
-				childVNode,
-				null,
-				null,
+			childVNode = newParentVNode._children[i] = createBackingNode(
+				createVNode(null, childVNode),
 				childVNode
 			);
 		} else if (Array.isArray(childVNode)) {
-			childVNode = newParentVNode._children[i] = createVNode(
-				Fragment,
-				{ children: childVNode },
-				null,
-				null,
+			childVNode = newParentVNode._children[i] = createBackingNode(
+				createVNode(Fragment, { children: childVNode }),
 				null
 			);
-		} else if (childVNode._dom != null || childVNode._component != null) {
```

Comment:
It's not possible to see a backing node as a child so we can safely remove this branch

Reply:
IMO this is the big sell of backing node in current Preact.